packages/cardano-metamask-snap/src/api/cardano__signTransaction/ui/collateral.ts

```diff
@@ -0,0 +1,52 @@
+import { heading, row, text } from '@metamask/snaps-sdk';
+
+import { getUiAccountIndex, section } from '../../ui';
+import { ADA_TICKER, assetValue } from './utils';
+import { OwnAddress, getAddressAccountIndex } from '../../address';
+
+export type CollateralReturnOutput = { address: string; coin: string } & (
+  | { isOwn: false }
+  | {
+      isOwn: true;
+      ownAddress: OwnAddress;
+    }
+);
+
+export type Collateral = {
+  collateralReturn: CollateralReturnOutput | undefined;
+  totalCollateral: string | undefined;
+};
+
+export const renderCollateralReturn = (
+  collateralReturnOutput: CollateralReturnOutput,
+) => {
+  const amountRow = row(
+    'Amount',
+    text(assetValue(collateralReturnOutput.coin, ADA_TICKER)),
+  );
+  const accountIndex = collateralReturnOutput.isOwn
+    ? getAddressAccountIndex(collateralReturnOutput.ownAddress)
+    : undefined;
+
+  if (accountIndex) {
+    return section([
+      heading(`Collateral return to ${getUiAccountIndex(accountIndex)}`),
+      amountRow,
+    ]);
+  }
+
+  return section([
```

Comment:
As discussed offline, I'd consider "packing" advanced info like this into a single text field, if that's the only way to have "expandable" content in the metamask prompt

Reply:
Ok, if you agree, lets finish what is currently open, I will then open a new PR which adds expandable info